odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1447,3 +1447,124 @@ def _validate_module_views(self, module):
                 view._check_xml()
             except Exception as e:
                 self.raise_view_error("Can't validate view:\n%s" % e, view.id)
+
+    def _fix_faulty_views(self, delete=False):
+        """ Use the _get_faulty_views and take appropriate actions to fix the issue """
+        for view in self:
+            view_id = view.id # allow to use even if view is deleted
+            faulty_views = view._get_faulty_views()
+            if faulty_views:
+                self._cr.execute('SAVEPOINT end_validation')
+                faulty_views._clean_properly(delete)
+                self._cr.execute("select count(id) from ir_ui_view where id = %s", (view_id,))
+                cnt = self.env.cr.fetchone()[0]
+                if cnt == 0 or view._is_valid():
+                    _logger.info("View %s can be fixed by disabling/deleting following views: %s" % (view_id,faulty_views))
+                    self._cr.execute('RELEASE end_validation')
+                else:
+                    self._cr.execute('ROLLBACK TO end_validation')
+                    self._cr.execute('RELEASE end_validation')
+                    _logger.warning("Issue occurs: The proposition made to fix the view %s (disable/delete %s) is not recommended" % (view,faulty_views))
+                    # should not happend
+
+    def _get_faulty_views(self,parent_view=None):
+        """ Returns the faulty views
```

Comment:
/python nazi on
PEP 257 for docstrings: https://www.python.org/dev/peps/pep-0257/
``` python
    def _get_faulty_views(self,parent_view=None):
        """Returns faulty views in the view's inheritance architecture.

            To determine the faulty views, we travel the tree recursively by first
            disabling all children, then re-enabling them one by one and check
            view validity. If there is an issue, the activated node or at least one
            of its children is faulty - disable faulty node and continue recursion.
    usage:
        view1._get_faulty_views()
        """
```

\+ adapt content of docstring if you use the thing i said below to find the parent automatically (even if the view on which this code is called is not a primary view)
/python nazi off

Reply:
I hope this is solved, I manually adapt it and also run a auto PEP8 on my code